core/src/main/java/org/geysermc/geyser/entity/type/Entity.java

```diff
@@ -412,6 +412,16 @@ public void setFlags(ByteEntityMetadata entityMetadata) {
         setInvisible((xd & 0x20) == 0x20);
     }
 
+    /**
+     * Ensure that collision is possible if this entity is a player. Should be overridden for players as their team
+     * might change the status of this.
```

Comment:
```suggestion
     * Tests whether an entity can be pushed by a player. For player entities, the team they're in determines if they can be pushed.
```
maybe something like this?